internal/pkg/aws/elbv2/elbv2.go

```diff
@@ -33,6 +35,44 @@ func New(sess *session.Session) *ELBV2 {
 	}
 }
 
+// ListenerRuleHostHeaders returns all the host headers for a listener rule.
+func (e *ELBV2) ListenerRuleHostHeaders(ruleARN string) ([]string, error) {
+	resp, err := e.client.DescribeRules(&elbv2.DescribeRulesInput{
+		RuleArns: aws.StringSlice([]string{ruleARN}),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("get listener rule for %s: %w", ruleARN, err)
+	}
+	if len(resp.Rules) == 0 {
+		// Ideally this should never happen.
+		return nil, fmt.Errorf("cannot find listener rule %s", ruleARN)
+	}
+	rule := resp.Rules[0]
+	hostHeaderSet := make(map[string]bool)
+	for _, condition := range rule.Conditions {
+		if aws.StringValue(condition.Field) == "host-header" {
+			for _, value := range condition.Values {
+				hostHeaderSet[aws.StringValue(value)] = true
+			}
+			if condition.HostHeaderConfig == nil {
+				// Ideally this should never happen.
+				continue
+			}
+			for _, value := range condition.HostHeaderConfig.Values {
+				hostHeaderSet[aws.StringValue(value)] = true
+			}
+			// Each rule can optionally include up to one of each of the conditions.
+			break
+		}
+	}
+	var hostHeaders []string
+	for hostHeader := range hostHeaderSet {
+		hostHeaders = append(hostHeaders, hostHeader)
+	}
+	sort.SliceStable(hostHeaders, func(i, j int) bool { return hostHeaders[i] < hostHeaders[j] })
```

Comment:
any reason for stable sort over plain `sort.Slice()`?

Reply:
It is just because we want to guarantee we always get the same output every time for the same input. However, here we could possibly use `Slice` because we only have one field for each element.